src/cdk/table/cell.ts

```diff
@@ -40,6 +40,14 @@ export class CdkColumnDef {
 
   /** @docs-private */
   @ContentChild(CdkHeaderCellDef) headerCell: CdkHeaderCellDef;
+
+  /**
+   * Returns the column's name with non-alphanumeric characters removed so it can be used
+   * as part of a CSS class. Allows for the special characters - and _.
+   */
+  getCssClassFriendlyName() {
+    return this.name.replace(/[^a-z0-9_-]/ig, '');
```

Comment:
Ah whoops, totally missed that you were storing the value into a separate variable. Makes sense